tensordict/tensordict.py

```diff
@@ -4902,6 +4926,78 @@ def _cat(
         return out
 
 
+@implements_for_lazy_td(torch.cat)
+def _lazy_cat(
+    list_of_tensordicts: Sequence[LazyStackedTensorDict],
+    dim: int = 0,
+    out: LazyStackedTensorDict | None = None,
+) -> LazyStackedTensorDict:
+    if not list_of_tensordicts:
+        raise RuntimeError("list_of_tensordicts cannot be empty")
+
+    batch_size = list(list_of_tensordicts[0].batch_size)
+    if dim < 0:
+        dim = len(batch_size) + dim
+    if dim >= len(batch_size):
+        raise RuntimeError(
+            f"dim must be in the range 0 <= dim < len(batch_size), got dim"
+            f"={dim} and batch_size={batch_size}"
+        )
+    stack_dim = list_of_tensordicts[0].stack_dim
+    if any([(td.stack_dim != stack_dim) for td in list_of_tensordicts]):
```

Comment:
```suggestion
    if any((td.stack_dim != stack_dim) for td in list_of_tensordicts):
```
![image](https://github.com/pytorch-labs/tensordict/assets/25529882/ffe62829-1e15-4d3f-81bf-6a1f27f833f2)

Reply:
oh i didn't know we could do that. absolutely, thanks!